lib/puppet/provider/service/launchd.rb

```diff
@@ -298,13 +317,21 @@ def enabled?
   # without actually loading/unloading the job.
   def enable
     overrides = self.class.read_plist(self.class.launchd_overrides)
-    overrides[resource[:name]] = { "Disabled" => false }
+    if self.class.get_os_version < 14
+      overrides[resource[:name]] = { "Disabled" => false }
+    else
+      overrides[resource[:name]] = false
+    end
     Plist::Emit.save_plist(overrides, self.class.launchd_overrides)
   end
 
   def disable
-    job_path, job_plist = plist_from_label(resource[:name])
-    job_plist["Disabled"] = true
-    Plist::Emit.save_plist(job_plist, job_path)
+    overrides = self.class.read_plist(self.class.launchd_overrides)
```

Comment:
This matches @keeleysam's changes in https://github.com/puppetlabs/puppet/pull/4164 to master.